src/RefreshToolbar/__tests__/RefreshToolbar.test.jsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import Button from 'src/Button/Button';
+import RefreshToolbar from '../RefreshToolbar';
+
+describe('<RefreshToolbar />', () => {
+    let count = 0;
+    let refreshing = false;
+    const handleRefresh = () => {
+        refreshing = true;
+        count += 1;
+    };
+
+    it('renders button as correctly-styled icon', () => {
+        const { container } = render((
+            <RefreshToolbar
+                handleRefresh={handleRefresh}
+                isRefreshing={refreshing}
+                refreshProgressLabel="Loading..."
+                refreshTooltipText="Refresh LogSync File Details"
+            />
+        ));
+
+        expect(container).toMatchSnapshot();
+    });
+
+    it('renders tooltip on hover', () => {
+        const { container } = render((
+            <RefreshToolbar
+                handleRefresh={handleRefresh}
+                isRefreshing={refreshing}
+                refreshProgressLabel="Loading..."
+                refreshTooltipText="Refresh LogSync File Details"
+            />
+        ));
+
+        fireEvent.mouseOver(container.querySelector('button'));
+
+        expect(container).toMatchSnapshot();
```

Comment:
If your tooltip comes into the page via a react portal outside your render scope, you can use `screen` to check the presence of the tooltip when hovered for a stronger test.

Reply:
checked this one    const tooltipAppears = screen.getByRole('tooltip');
        expect(tooltipAppears).toBeInTheDocument();
But the tooltip presence cannot be detected. The other way is to check the tooltip text but we do not have tooltip text in our functionality.

Reply:
It looks like your snapshot does not have anything really rendered for tooltip, and you're passing in the 'refreshTooltipText' prop which is not in the interface nor used in the component, so I would expect that you can't find a tooltip via the test if there is no tooltip to find. I'm not sure the purpose of this test if there is no tooltip functionality in the component.